src/tiles/text-snippet-block.ts

```diff
@@ -0,0 +1,126 @@
+import {
+  css,
+  CSSResultGroup,
+  html,
+  LitElement,
+  nothing,
+  TemplateResult,
+} from 'lit';
+import { customElement, property } from 'lit/decorators.js';
+import { join } from 'lit/directives/join.js';
+
+@customElement('text-snippet-block')
+export class TextSnippetBlock extends LitElement {
+  @property({ type: Array }) snippets?: string[] = [];
+
+  @property({ type: String }) viewSize: string = 'desktop';
+
+  render() {
+    const viewSizeClass = this.viewSize === 'grid' ? 'grid' : 'list';
+
+    if (!this.snippets || this.snippets.length === 0) return html`${nothing}`;
+
+    return html`
+      <div id="container" class=${viewSizeClass}>
+        ${this.ellipsisJoinedSnippets}
+      </div>
+
+      ${this.viewSize === 'grid' ? html`<div id="separator"></div>` : nothing}
+    `;
+  }
+
+  /**
+   * An array of HTML templates derived from the snippets, with ellipses inserted
+   * at the beginning, end, and between each pair of snippets.
+   */
+  private get ellipsisJoinedSnippets(): TemplateResult {
+    return html`&hellip; ${join(this.snippetTemplates, html` &hellip; `)}
+    &hellip;`;
```

Comment:
q: was the line break because of max-line length rule? perhaps we move the whole `html` block under the return so it can be 1 line..

Reply:
Yeah, I think this was Prettier's doing. Will fix the formatting here.